File: src/model/course-admin/CourseLessonRequest.ts
import { LessonType } from "@/types/enum/LessonType";
import { TemplateSelection } from "./TemplateSelection";

export interface CourseLessonRequest {
  lessonId?: string;
  moduleId?: string;
  creatorId?: string;
  title: string;
  description?: string;
  content?: string;
  videoUrl?: string;
  videoFile?: File;
  duration?: number;
  orderIndex: number;
  isPreview: boolean;
  isPublished: boolean;
  slug?: string;
  lessonType: LessonType;

  exerciseTemplateIds?: string[];
  quizTemplateIds?: string[];

  // Store template info for UI display (not sent to backend)
  exerciseTemplates?: TemplateSelection[];
  quizTemplates?: TemplateSelection[];
}

File: src/main/java/com/truongsonkmhd/unetistudy/dto/lesson_dto/CourseLessonRequest.java
package com.truongsonkmhd.unetistudy.dto.lesson_dto;

import com.fasterxml.jackson.annotation.JsonProperty;
import com.truongsonkmhd.unetistudy.common.LessonType;
import lombok.*;
import lombok.experimental.FieldDefaults;

import java.util.List;
import java.util.UUID;

@Data
@NoArgsConstructor
@AllArgsConstructor
@Builder
@FieldDefaults(level = AccessLevel.PRIVATE)
public class CourseLessonRequest {

     UUID lessonId; // update thì gửi, create thì null
     UUID moduleId;
     UUID creatorId;

     String title;
     String description;
     String content;
     String videoUrl; // YouTube URL, ví dụ: https://www.youtube.com/watch?v=XXX

     Integer orderIndex;

     @JsonProperty("isPreview")
     Boolean isPreview;
     @JsonProperty("isPublished")
     Boolean isPublished;

     String slug;
     LessonType lessonType;

     Integer totalPoints;

     List<UUID> exerciseTemplateIds;
     List<UUID> quizTemplateIds;
}
